### ISSUE TITLE
Error in Irene

### ISSUE THREAD

**Issue by jmbenlloch** [next-exp/IC#81]
I have run Irene on MC Kr at CERN, but 41 of the jobs had errors during processing. All of them are like this:

```
Opening ./ic_dst_NEXT_v0_08_06_Kr_ACTIVE_7_0_7bar_MCRD_10000.root.h5... Events in file = 10000
# PMT                  => 12
# SiPM                 => 1792
PMT WL                 => 48000
SIPM WL                => 1200
event in file = 99, total = 100
error
Traceback (most recent call last):
  File "/afs/cern.ch/work/j/jobenllo/software/IC/invisible_cities/cities/irene.py", line 548, in <module>
    IRENE(sys.argv)
  File "/afs/cern.ch/work/j/jobenllo/software/IC/invisible_cities/cities/irene.py", line 540, in IRENE
    nevt = fpp.run(nmax=nevts, store_pmaps=True)
  File "/afs/cern.ch/work/j/jobenllo/software/IC/invisible_cities/cities/irene.py", line 429, in run
    rebin  = False)
  File "invisible_cities/core/peak_functions_c.pyx", line 105, in invisible_cities.core.peak_functions_c.find_S12 (invisible_cities/core/peak_functions_c.c:5121)
  File "invisible_cities/core/peak_functions_c.pyx", line 134, in invisible_cities.core.peak_functions_c.find_S12 (invisible_cities/core/peak_functions_c.c:4232)
IndexError: Out of bounds on buffer access (axis 0)
```

I have checked the logs to find a file in which the error appears soon, in particular with:

`lxplus.cern.ch:/eos/experiment/next/dataic/DiomiraMS/Kr/ic_dst_NEXT_v0_08_06_Kr_ACTIVE_7_0_7bar_MCRD_10000.root.h5`

The error occurs before event 200. The config file I am using is this one:

```
PATH_IN .
PATH_OUT .
FILE_IN ic_dst_NEXT_v0_08_06_Kr_ACTIVE_7_0_7bar_MCRD_10000.root.h5
FILE_OUT pmaps_dst_NEXT_v0_08_06_Kr_ACTIVE_7_0_7bar_MCRD_10000.root.h5
COMPRESSION ZLIB4

RUN_NUMBER 0
NPRINT 100

NBASELINE 38000
THR_TRIGGER 5

NMAU 100
THR_MAU 3
THR_CSUM 0.5
S1_TMIN 10
S1_TMAX 590
S1_STRIDE 4
S1_LMIN 10
S1_LMAX 20
S2_TMIN 10
S2_TMAX 1190
S2_STRIDE 40
S2_LMIN 100
S2_LMAX 100000
THR_ZS 5
THR_SIPM_S2 25

NEVENTS 20
RUN_ALL True
```

Any idea @jjgomezcadenas ?

**Comment by jjgomezcadenas** [comment]
There is something really funny… you are opening a MCRD file

> Opening ./ic_dst_NEXT_v0_08_06_Kr_ACTIVE_7_0_7bar_MCRD_10000.root.h5..
But for Irene you should be opening a RWF file (the output of DIOMIRA)…

Notice that we don’t re-write RWF files in MCRD files any more…

**Comment by jmbenlloch** [comment]
That is a naming mistake in my production script. The original hdf5 file from ART (MCRD) was dst_NEXT_v0_08_06_Kr_ACTIVE_0_0_7bar_MCRD_10000.root.h5, as you can see here:

```
$ h5ls -r dst_NEXT_v0_08_06_Kr_ACTIVE_0_0_7bar_MCRD_10000.root.h5
/                        Group
/Detector                Group
/Detector/DetectorGeometry Dataset {1}
/MC                      Group
/MC/MCTracks             Dataset {68986/Inf}
/Run                     Group
/Run/events              Dataset {10000/Inf}
/Run/runInfo             Dataset {1}
/Sensors                 Group
/Sensors/DataBLR         Dataset {12}
/Sensors/DataPMT         Dataset {12}
/Sensors/DataSiPM        Dataset {1792}
/pmtrd                   Dataset {10000/Inf, 12, 1200000}
/sipmrd                  Dataset {10000/Inf, 1792, 1200}
```

The output of diomira is written with the same file name starting with "ic_". Therefore, the file I'm opening is actually an RWF:

```
$ h5ls -r ic_dst_NEXT_v0_08_06_Kr_ACTIVE_0_0_7bar_MCRD_10000.root.h5 
/                        Group
/MC                      Group
/MC/FEE                  Dataset {1/Inf}
/RD                      Group
/RD/pmtblr               Dataset {10000/Inf, 12, 48000}
/RD/pmtrwf               Dataset {10000/Inf, 12, 48000}
/RD/sipmrwf              Dataset {10000/Inf, 1792, 1200}
```

I'll fix the name issue, but it should not be related to this Irene issue.

**Comment by jjgomezcadenas** [comment]
Since Irene runs on data, the only reason I can come up with is a problem with vector dimensions in the MC (e.g, a badly conformed event). Can you isolate one event that fails, so that we can look at it?

**Comment by jmbenlloch** [comment]
I have isolated in one hdf5 file one of those failing events. You can get the file here: http://jobenllo.web.cern.ch/jobenllo/irene_bug_Kr_ACTIVE_7bar_RWF.h5

**Comment by jacg** [comment]
I trust that you gentlemen (@jmbenlloch & @jjgomezcadenas) are planning to write an automated test that indicates the presence of this bug, *before* fixing it! Assuming it really is a bug. And that you are planning on leaving the test in the repo once the bug is fixed.

**Comment by jjgomezcadenas** [comment]
Ay, ay sir. I am not sure if the bug is a bug (in Irene) or a bug in the simulation (which may come from Diomira or upstream, from the simulation), but for sure, a test will be written!

**Comment by jjgomezcadenas** [comment]
A very interesting case. The BLR waveforms are empty and the RWF are just “zero” (on top of the baseline shift). Can you find the same event in the MCRD file? I believe that the problem comes from NEXUS, not from Diomira, but we need to check.

If the problem is an “empty event”, as I believe, then Is easy to solve, we just need to impose that there is a minimum of energy in the PMTs before proceeding. There is a clear associated test as proposed by @jacg, which throws “unexpected” (e.g, null) waveforms to Irene. Irene should be able to handle those kind of monsters.

**Comment by jmbenlloch** [comment]
This is the same event before running Diomira: http://jobenllo.web.cern.ch/jobenllo/irene_bug_Kr_ACTIVE_7bar_MCRD.h5

**Comment by jmbenlloch** [comment]
Solved by #98.
